docs(zone.js): update docs to enable beforeunload (#57881)
angular/angular: packages/zone.js/MODULE.md

|handleUnhandledPromiseRejection|NodeJS handle unhandledPromiseRejection from ZoneAwarePromise|__Zone_disable_handleUnhandledPromiseRejection = true|
|crypto|NodeJS patch crypto function as macroTask|__Zone_disable_crypto = true|

### Test Framework

|Module Name|Behavior with zone.js patch|How to disable|
|--|--|--|
|Jasmine|Jasmine APIs patch|__Zone_disable_jasmine = true|
|Mocha|Mocha APIs patch|__Zone_disable_mocha = true|

### `on` properties

You can also disable specific `on` properties by setting `__Zone_ignore_on_properties` as follows. For example, if you want to disable `window.onmessage` and `HTMLElement.prototype.onclick` from zone.js patching, you can do so like this:

```html
<script>
  __Zone_ignore_on_properties = [
    {
      target: window,
      ignoreProperties: ['message'],
    },
    {
      target: HTMLElement.prototype,
      ignoreProperties: ['click'],
    },
  ];
</script>
<script src="../bundles/zone.umd.js"></script>
```

> Ignoring `on` properties from being patched means that callbacks will always be invoked within the root context, regardless of where the `on` callback has been set. Even if `onclick` is set within a child zone, the callback will be called inside the root zone:

Reviewer: ```suggestion
Excluding `on` properties from being patched means that callbacks will always be invoked within the root context, regardless of where the `on` callback has been set. Even if `onclick` is set within a child zone, the callback will be called inside the root zone:
```

```ts
Zone.current.fork({ name: 'child' }).run(() => {
  document.body.onclick = () => {
    console.log(Zone.current); // <root>
  };
});
```

> For unpatching events being added through `addEventListener`, please refer to [UnpatchedEvents](./STANDARD-APIS.md#unpatchedevents).

Reviewer: ```suggestion
You can find more information on adding unpatched events via `addEventListener`, please refer to [UnpatchedEvents](./STANDARD-APIS.md#unpatchedevents).
```

### Error

> By default, `zone.js/plugins/zone-error` will not be loaded due to performance concerns.

Reviewer: ```suggestion
By default, `zone.js/plugins/zone-error` will not be loaded for performance reasons.
```

This package provides the following functionality:

1. **Error Inheritance:** Handle the `extend Error` issue:

   ```ts
     class MyError extends Error {}
     const myError = new MyError();
     console.log('is MyError instanceof Error', (myError instanceof Error));
   ```

>    Without the `zone-error` patch, the example above will output `false`; with the patch, the result will be `true`.

Reviewer: ```suggestion
   Without the `zone-error` patch, the example above will output `false`. With the patch, the result will be `true`.
```
